File: backend/java/backend_java_spring/src/main/java/dev/reference/app/service/ItemService.java
package dev.reference.app.service;

import dev.reference.app.dto.HealthResponse;
import dev.reference.app.dto.ItemDto;
import dev.reference.app.dto.ItemsResponse;
import dev.reference.app.repository.ItemRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ItemService {

    private final ItemRepository repository;

    public ItemService(ItemRepository repository) {
        this.repository = repository;
    }

    public HealthResponse health() {
        return new HealthResponse("ok", "reference-app-copy");
    }

    public ItemsResponse listItems() {
        List<ItemDto> items = repository.findAll().stream()
                .map(entity -> new ItemDto(entity.getId(), entity.getName(), entity.getDescription()))
                .toList();
        return new ItemsResponse(items, "postgresql");
    }
}

File: backend/kotlin/backend-kotlin-spring/src/main/kotlin/dev/reference/app/service/ItemService.kt
package dev.reference.app.service

import dev.reference.app.dto.HealthResponse
import dev.reference.app.dto.ItemDto
import dev.reference.app.dto.ItemsResponse
import dev.reference.app.repository.ItemRepository
import org.springframework.data.domain.Sort
import org.springframework.stereotype.Service

@Service
class ItemService(
    private val repository: ItemRepository,
) {
    fun health(): HealthResponse = HealthResponse("ok", SERVICE_NAME)

    fun listItems(): ItemsResponse {
        val items = repository.findAll(BY_ID).map { entity ->
            ItemDto(requireNotNull(entity.id), entity.name, entity.description)
        }
        return ItemsResponse(items, "postgresql")
    }

    companion object {
        /** Matches `health_service` for this module in `deploy/matrix.yaml`. */
        private const val SERVICE_NAME = "backend-kotlin-spring"

        private val BY_ID: Sort = Sort.by(Sort.Direction.ASC, "id")
    }
}
